launch_testing_ros/test/examples/wait_for_topic_launch_test.py

```diff
@@ -0,0 +1,91 @@
+# limitations under the License.
+
+import os
+import sys
+import unittest
+
+import launch
+import launch.actions
+import launch_ros.actions
+import launch_testing.actions
+import launch_testing.markers
+from launch_testing_ros import WaitForTopics
+import pytest
+from std_msgs.msg import String
+
+
+def generate_node(i):
+    """Return node and remap the topic based on the index provided."""
+    path_to_test = os.path.dirname(__file__)
+    return launch_ros.actions.Node(
+        executable=sys.executable,
+        arguments=[os.path.join(path_to_test, 'talker.py')],
+        name='demo_node_' + str(i),
+        additional_env={'PYTHONUNBUFFERED': '1'},
+        remappings=[('chatter', 'chatter_' + str(i))]
+    )
+
+
+@pytest.mark.launch_test
+@launch_testing.markers.keep_alive
+def generate_test_description():
+    # 'n' changes the number of nodes and topics generated for this test
+    n = 5
+    description = [generate_node(i) for i in range(n)] + [launch_testing.actions.ReadyToTest()]
+    return launch.LaunchDescription(description), {'count': n}
+
+
+class TestFixture(unittest.TestCase):
+
+    def test_topics_successful(self, count):
+        """All the supplied topics should be read successfully."""
+        topic_list = [('chatter_' + str(i), String) for i in range(count)]
+        expected_topics = {'chatter_' + str(i) for i in range(count)}
+
+        # Method 1 : Using the magic methods and 'with' keyword
+        wait_for_node_object_1 = WaitForTopics(topic_list, timeout=2.0)
+        with wait_for_node_object_1:
+            assert wait_for_node_object_1.topics_received() == expected_topics
+            assert wait_for_node_object_1.topics_not_received() == set()
+
+        # Multiple instances of WaitForNode() can be created safely as
+        # their internal nodes spin in separate contexts
+        # Method 2 : Manually calling wait() and shutdown()
+        wait_for_node_object_2 = WaitForTopics(topic_list, timeout=2.0)
+        assert wait_for_node_object_2.wait()
+        assert wait_for_node_object_2.topics_received() == expected_topics
+        assert wait_for_node_object_2.topics_not_received() == set()
+        wait_for_node_object_2.shutdown()
+
+    def test_topics_unsuccessful(self, count):
+        """All topics should be read except for the 'invalid_topic'."""
+        topic_list = [('chatter_' + str(i), String) for i in range(count)]
+        # Add a topic that will never have anything published on it
+        topic_list.append(('invalid_topic', String))
+        expected_topics = {'chatter_' + str(i) for i in range(count)}
+
+        # Method 1
+        wait_for_node_object_1 = WaitForTopics(topic_list, timeout=2.0)
+        with pytest.raises(RuntimeError):
+            with wait_for_node_object_1:
+                assert wait_for_node_object_1.topics_received() == expected_topics
+                assert wait_for_node_object_1.topics_not_received() == {'invalid_topic'}
```

Comment:
Since we expect the with-clause to raise, the assert on topics (not) received should not be executed:
```suggestion
        with pytest.raises(RuntimeError):
            with WaitForTopics(topic_list, timeout=2.0):
                pass
```